src/Symfony/Component/Console/Input/InputInterface.php

```diff
@@ -122,7 +122,7 @@ public function getOptions();
      *
      * @param string $name The option name
      *
-     * @return string|string[]|bool|null The option value
+     * @return string|string[]|int|bool|null The option value
```

Comment:
Thank for this PR, but this change looks wrong,  The options can be:
- `string`: `--option=foo`
- `string[]`: `--option=foo --option=bar`
- `bool`:  `--option` + `VALUE_NONE`
- `null`:  `--option` + `VALUE_OPTIONAL`

But there is no case where the value can be an integer.

Note: `--option=123` returns the string `"123"`.

Reply:
Indeed, see also https://github.com/symfony/symfony/pull/28374#issuecomment-419815248.

Reply:
We should rather remove `int` from the `getDefault()` doc block.

Reply:
`InputOption` declares `int` as a valid type for the default value: in the constructor signature as well as for `getDefault()` and `setDefault()`. This looks like a mistake to me because you can never produce an integer on the command line. But it means that it is possible, when following the documented public API, to receive an integer from `getOption()`.

Reply:
Indeed when the default value is used then it can be the case that `getOption` returns an integer. I just tested it quickly.

```php
public function configure(): void
{
    $this->addOption('foo', null, InputOption::VALUE_REQUIRED, '', 123);
}

protected function execute(InputInterface $input, OutputInterface $output): int
{
    var_dump($input->getOption('foo'));die;
}
```

And then executed without passing any option at all:
==> `int(123)`

Reply:
Would you mind to change this PR to fix the `default` signature? @oliverklee